src/senselab/audio/tasks/quality_control/metrics.py

```diff
@@ -0,0 +1,744 @@
+
+    # Get peak signal amplitude from time-domain
+    peak = np.max(np.abs(waveform))
+
+    return float(20 * np.log10(peak / noise_rms))
+
+
+def amplitude_skew_metric(audio: Audio) -> float:
+    """Calculates the skew of the audio signal amplitude.
+
+    Args:
+        audio (Audio): The SenseLab Audio object.
+
+    Returns:
+        float: Skew of the flattened amplitude distribution.
+    """
+    waveform = audio.waveform
+    assert waveform.ndim == 2, "Expected waveform shape (num_channels, num_samples)"
+    return float(scipy.stats.skew(waveform.flatten().numpy()))
+
+
+def amplitude_kurtosis_metric(audio: Audio) -> float:
+    """Calculates the kurtosis of the audio signal amplitude.
+
+    Args:
+        audio (Audio): The SenseLab Audio object.
+
+    Returns:
+        float: Kurtosis of the flattened amplitude distribution.
+    """
+    waveform = audio.waveform
+    assert waveform.ndim == 2, "Expected waveform shape (num_channels, num_samples)"
+    return float(scipy.stats.kurtosis(waveform.flatten().numpy()))
+
+
+def amplitude_interquartile_range_metric(audio: Audio) -> float:
+    """Calculates the interquartile range (IQR) of the audio signal amplitude.
+
+    Args:
+        audio (Audio): The SenseLab Audio object.
+
+    Returns:
+        float: IQR of the flattened amplitude distribution.
+    """
+    waveform = audio.waveform
+    assert waveform.ndim == 2, "Expected waveform shape (num_channels, num_samples)"
+    return float(scipy.stats.iqr(waveform.flatten().numpy()))
+
+
+def phase_correlation_metric(audio: Audio, frame_length: int = 2048, hop_length: int = 512) -> float:
+    """Computes average inter-channel correlation for stereo or multi-channel audio."""
+    waveform = audio.waveform
+
+    if waveform.ndim != 2:
+        return 1.0  # Treat 1D as mono
+
+    num_channels, num_samples = waveform.shape
+    if num_channels < 2:
+        return 1.0  # Mono
+
+    if num_samples < frame_length:
+        frame_length = num_samples
+        hop_length = max(1, num_samples // 2)
+
+    correlation_values = []
+
+    for i in range(0, num_samples - frame_length + 1, hop_length):
+        frame = waveform[:, i : i + frame_length]
+        if frame.shape[1] < 2:
+            continue
+
+        corr_matrix = np.corrcoef(frame.numpy())
+        upper_triangle = corr_matrix[np.triu_indices(num_channels, k=1)]
+
+        # Exclude NaNs (e.g. from silent channels)
+        valid_corrs = upper_triangle[~np.isnan(upper_triangle)]
+        if valid_corrs.size > 0:
+            correlation_values.append(np.mean(valid_corrs))
+
+    return float(np.mean(correlation_values)) if correlation_values else 0.0
+
+
+def primary_speaker_ratio_metric(audio: Audio) -> float:
+    """Calculates the ratio of the primary speaker's duration to the total duration.
+
+    The primary speaker ratio is computed from speaker diarization results.
+    If diarization is not available in audio.metadata, it will be computed automatically.
```

Comment:
I don't think we've ever used metadata like this previously. I think it's a good idea, but we should try to standardize our practices with it